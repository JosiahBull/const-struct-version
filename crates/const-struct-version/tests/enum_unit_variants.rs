#![allow(dead_code)]

use const_struct_version::StructVersion;

#[derive(StructVersion)]
enum EnumUnitVariants {
    A,
    B,
}

#[derive(StructVersion)]
enum EnumUnitVariants2 {
    B,
    A,
}

#[test]
fn test_enum_unit_variants_changes_hash() {
    let version = <EnumUnitVariants as StructVersion>::version();
    let version2 = <EnumUnitVariants2 as StructVersion>::version();

    insta::assert_debug_snapshot!(version);
    insta::assert_debug_snapshot!(version2);

    // Ensure all are different.
    assert_ne!(version, version2);
}
