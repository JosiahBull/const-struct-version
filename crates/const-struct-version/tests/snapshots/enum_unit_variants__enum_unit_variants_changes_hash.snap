---
source: crates/const-struct-version/tests/enum_unit_variants.rs
expression: version
---
"06d945942aa26a61be18c3e22bf19bbca8dd2b5d"
